clap_derive/tests/subcommands.rs

```diff
@@ -161,47 +161,52 @@ fn test_tuple_commands() {
     assert!(!output.contains("Not shown"));
 }
 
-#[test]
```

Comment:
Let's create an issue to rethink about Flatten and Composability.

Reply:
Mind explaining what you mean by "Composability"?

Reply:
I am thinking of the future. I actually started looking into Clap because I was working on a [higher level CLI framework](https://github.com/termapps/oclif). In this, I wanted people to make plugins and CLI to use them. For example, I wanted to provide a plugin which provides a group of subcommands that I want a CLI to have at top-level, I would want to use `flatten`. But this is not the only scenario.

What I am saying is we want to make sure we support all kinds of things people want to do if they can share Args and Apps.